docs/aggregation.ipynb

```diff
@@ -1,260 +1,294 @@
-    "metadata": {},
-    "source": [
-     "Notice how existing dimensions on each datasets are unchanged by the transformation. If you want to average over existing dimension, you would need to do that aggregation yourself, e.g., by averaging inside each chunk before combining the data."
-    ]
-   },
-   {
-    "cell_type": "markdown",
-    "id": "6ab152b6",
-    "metadata": {},
-    "source": [
-     "Similarly, the keys fed into `xbeam.Mean.PerKey` can be any hashables, including but not limited to `xbeam.Key`:"
-    ]
-   },
-   {
-    "cell_type": "code",
-    "execution_count": 2,
-    "id": "b2399483",
-    "metadata": {},
-    "outputs": [
-     {
-      "name": "stderr",
-      "output_type": "stream",
-      "text": [
-       "WARNING:apache_beam.options.pipeline_options:Discarding unparseable args: ['/Users/shoyer/miniconda3/envs/xarray-beam/lib/python3.9/site-packages/ipykernel_launcher.py', '-f', '/Users/shoyer/Library/Jupyter/runtime/kernel-e19c251a-dc50-467c-aaed-1252152cd403.json']\n"
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "aaf2f2cb",
+   "metadata": {},
+   "source": [
+    "# Aggregation\n",
+    "\n",
+    "Xarray-Beam can perform efficient distributed data aggregation in the \"map-reduce\" model. \n",
+    "\n",
+    "This currently only includes `Mean`, but we would welcome contributions of other aggregation functions such as `Sum`, `Std`, `Var`, `Min`, `Max`, etc."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4b77b953",
+   "metadata": {},
+   "source": [
+    "## High-level API\n",
+    "\n",
+    "The `Mean` transformation comes in three forms:  {py:class}`Mean <xarray_beam.Mean>`, {py:class}`Mean.Globally <xarray_beam.Mean.Globally>`, and {py:class}`Mean.PerKey <xarray_beam.Mean.PerKey>`. The implementation is highly scalable, based on a Beam's [`CombineFn`](https://beam.apache.org/documentation/transforms/python/aggregation/combineglobally/#example-4-combining-with-a-combinefn)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39719ac3",
+   "metadata": {},
+   "source": [
+    "The high-level `Mean` transform can be used to aggregate a distributed dataset across an existing dimension or dimensions, similar to Xarray's `.mean()` method:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "68d6f586",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import apache_beam as beam\n",
+    "import numpy as np\n",
+    "import xarray_beam as xbeam\n",
+    "import xarray\n",
+    "\n",
+    "ds = xarray.tutorial.load_dataset('air_temperature')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "f18bb1ff",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in '[6]: Mean/PerKey/CombinePerKey(MeanCombineFn)/GroupByKey'. \n",
```

Comment:
In the new Combiner class itself, is it possible to implement something s.t. we don't get these warnings?

Reply:
I bet it has something to do with making xbeam.Key serializable to Beam...

Reply:
I agree, this would be nice. I don't know how to fix this. I believe you can register a Coder for specific types, but I vaguely recall that it didn't handle nested types like `tuple[xbeam.Key, xarray.Dataset]`, at least without specific annotations.

Reply:
Filed an issue about this in https://github.com/google/xarray-beam/issues/77